IB/ipoib: set max CM MTU when moving to CM mode

This will releive the user from the need to restore CM mode MTU
every time we switch from UD to CM mode.

Signed-off-by: Eli Cohen <eli@mellanox.co.il>
Signed-off-by: Vladimir Sokolovsky <vlad@mellanox.co.il>

---
 drivers/infiniband/ulp/ipoib/ipoib_cm.c |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

Index: ofed_kernel/drivers/infiniband/ulp/ipoib/ipoib_cm.c
===================================================================
--- ofed_kernel.orig/drivers/infiniband/ulp/ipoib/ipoib_cm.c
+++ ofed_kernel/drivers/infiniband/ulp/ipoib/ipoib_cm.c
@@ -1459,9 +1459,14 @@ static ssize_t set_mode(struct device *d
 
 		rtnl_lock();
 		dev->features &= ~(NETIF_F_IP_CSUM | NETIF_F_SG | NETIF_F_TSO);
-		rtnl_unlock();
 		priv->tx_wr.send_flags &= ~IB_SEND_IP_CSUM;
 
+		if (ipoib_cm_max_mtu(dev) > priv->mcast_mtu)
+			ipoib_warn(priv, "mtu > %d will cause multicast packet drops.\n",
+				   priv->mcast_mtu);
+		dev_set_mtu(dev, ipoib_cm_max_mtu(dev));
+		rtnl_unlock();
+
 		ipoib_flush_paths(dev);
 		return count;
 	}
